### PR TITLE
newaxis

### PR DESCRIPTION
**Pull request by mcabbott**
After:
```
julia> const newaxis = [CartesianIndex()];

julia> rand(2,2)[:, newaxis, :] |> size
(2, 1, 2)

julia> NamedDimsArray(rand(2,2), (:a,:b))[:, newaxis, :] |> names
(:a, :_, :b)
```

### CODE REVIEW

**Comment by oxinabox** [comment]
I have not reviewed this yet

**Comment by mcabbott** [comment]
No rush. Another test case before I forget… which doesn’t error, but why should it have name `:x` not `:z`, perhaps it should have neither?
```
julia> nda = NamedDimsArray(rand(1:9, 3,3,3), (:x, :y, :z));

julia> nda[rand(3,3,3) .> 0.5]
17-element NamedDimsArray{(:x,),Int64,1,Array{Int64,1}}:
```

**Comment by oxinabox** [comment]
> Another test case before I forget… which doesn’t error, but why should it have name :x not :z, perhaps it should have neither?

Yeah, I think it should have neither. Maybe open issue for that one?

File: src/name_core.jl [hunk 1]
```diff
@@ -231,13 +225,23 @@ Dimensions indexed with scalars are dropped
 @generated function remaining_dimnames_from_indexing(dimnames::Tuple, inds)
     # 0-Allocation see:
     # `@btime (()->remaining_dimnames_from_indexing((:a, :b, :c), (:,390,:)))()``
```
[the remark this answers is not in the document]
Comment (mcabbott) [reply src/name_core.jl]: Yes I did:
```
julia> @btime (()->remaining_dimnames_from_indexing((:a, :b, :c), (:,390,:)))()
  1.420 ns (0 allocations: 0 bytes)
(:a, :c)
```

File: test/wrapper_array.jl [hunk 2]
```diff
@@ -52,9 +54,24 @@ end
...
+
+        @test names(nda[:,newaxis,:]) == (:x, :_, :y)
+        @test size(nda[:,newaxis,:]) == (2, 1, 2)
+
+        @test names(nda[1,newaxis,1]) == (:_,)
+        @test size(nda[1,newaxis,1]) == (1,)
+
+        @test names(nda[CartesianIndex(1,1),newaxis]) == (:_,)
+        @test size(nda[CartesianIndex(1,1),newaxis]) == (1,)
+        @test nda[CartesianIndex(1,1),newaxis][1] == nda[1,1]
```
Comment (nickrobinson251) [inline test/wrapper_array.jl]: minor style point: would be good to stick to the Blue style of spaces after commas throughout this block

**Comment by oxinabox** [comment]
Thanks!
Can you bump the patch version so I can tag a release after I merge

**Comment by oxinabox** [comment]
Thanks

**Pull request merged**